crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -795,10 +796,20 @@ impl CheckpointExecutor {
                             )
                             .expect("failed to acquire shared version assignments");
 
-                        let env = ExecutionEnv::new()
+                        let mut env = ExecutionEnv::new()
                             .with_assigned_versions(assigned_versions)
                             .with_expected_effects_digest(*expected_fx_digest);
 
+                        // Check if the expected effects indicate insufficient balance
+                        if let ExecutionStatus::Failure { error, .. } = effects.status() {
+                            if matches!(
+                                error,
+                                ExecutionFailureStatus::InsufficientBalanceForWithdraw
```

Comment:
you can collapse this into `if let ExecutionStatus::Failure { error: ExecutionFailureStatus::InsufficientBalanceForWithdraw, .. }`